DISTPG-364 Backup and disaster recovery

nastena1606 opened the pull request:
modified:   docs/solutions/backup-recovery.md
	modified:   docs/solutions/dr-pgbackrest-setup.md

patrickbirch reviewed (approved):
LGTM

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+- High availability ensures guaranteed service levels at all times. This solution involves configuring one or more standby systems to an active database, and the ability to switch seamlessly to that standby when the primary database becomes unavailable, for example, during a power outage or a server crash. To learn more about high-availability solutions with Percona Distribution for PostgreSQL, r ...
+- Disaster Recovery protects the database instance against accidental or malicious data loss or data corruption. Disaster recovery can be achieved by using either the options, provided by PostgreSQL, or external extensions. 
+
+    ??? "PostgreSQL disaster recovery options"
```
I presume that "???" is either a typo or some unprintable characters that you might want to remove.

Also, this should probably be a subheading.

nastena1606 replied:
"???" is the formatting for the collapsible section, collapsed by default. 

Amended padding to make it a separate subsection

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+A Disaster Recovery (DR) solution ensures that a system can be quickly restored to a normal operational state if something unexpected happens. When operating a database, you would back up the data as frequently as possible and have a mechanism to restore that data when needed. Disaster Recovery is often mistaken for high availability (HA), but they are two different concepts altogether:
+
+- High availability ensures guaranteed service levels at all times. This solution involves configuring one or more standby systems to an active database, and the ability to switch seamlessly to that standby when the primary database becomes unavailable, for example, during a power outage or a server crash. To learn more about high-availability solutions with Percona Distribution for PostgreSQL, r ...
+- Disaster Recovery protects the database instance against accidental or malicious data loss or data corruption. Disaster recovery can be achieved by using either the options, provided by PostgreSQL, or external extensions. 
```
Remove the "," after options IMHO

nastena1606 replied:
Done

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+
+        PostgreSQL offers multiple options for setting up database disaster recovery. 
+
+        - **[pg_dump](https://www.postgresql.org/docs/11/app-pgdump.html) or the [pg_dumpall](https://www.postgresql.org/docs/11/app-pg-dumpall.html) utilities**
```
IMHO, we should be providing links to the latest PG documentation. Right now, some links point to 11 and some to current.

nastena1606 replied:
Fixed

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+
+        - **PostgreSQL [pg_basebackup](https://www.postgresql.org/docs/11/app-pgbasebackup.html)**
+
+        This backup tool is provided by PostgreSQL. It uses the [PostgreSQL Write Ahead logs](https://www.postgresql.org/docs/current/wal-intro.html) (WAL) and can thus back up data even when the database instance is running. 
```
This is factually incorrect. Please see the pg_basebackup documentation

nastena1606 replied:
Amended

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+
+The following diagram illustrates the architecture layout:
+
+![pgBackRest Implementation Architecture](../_images/diagrams/DR-architecture.png)
```
This link seems to be broken

nastena1606 replied:
This is the markdown formatting. The image is a separate file and renders in the html output. You can see the test deployment here: https://new.percona.com/doc/postgresql/11/solutions/backup-recovery.html (SSO login is required)

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+Finally, `pgBackRest` also supports restoring PostgreSQL databases to a different PostgreSQL instance or a separate data directory.
+
+
+## System architecture
```
This seems a little too abrupt. Perhaps some leading text explaining what this section is for will be helpful.

nastena1606 replied:
Added an intro section

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+
+In our architecture, `pgBackRest` is located on a remote host and the nodes communicate with each other via passwordless SSH.
+
+!!! note
```
Is "!!!" expected here?

nastena1606 replied:
Yes, it marks a note and will be rendered as an admonition in the output.

hqakhtar commented on docs/solutions/dr-pgbackrest-setup.md:
```diff
@@ -0,0 +1,481 @@
+
+As the example configuration, we will use the nodes with the following IP addresses:
+
+| **Node name**   | **Internal IP address** |
```
Would be nice to have the same ordering of these nodes as mentioned in the "Components" section.

nastena1606 replied:
Fixed

hqakhtar commented on docs/solutions/backup-recovery.md:
```diff
@@ -0,0 +1,81 @@
+- `pg-repo` is the remote backup repository and hosts `pgBackRest`. It’s important to host the backup repository on a physically separate instance, to be accessed when the target goes down. 
+- `pg-secondary` is the _secondary_ PostgreSQL node. Don't confuse it with a hot standby. "Secondary" in this context means a PostgreSQL instance that’s idle. We will restore the database backup to this instance when the primary PostgreSQL instance goes down. 
+
+In our architecture, `pgBackRest` is located on a remote host and the nodes communicate with each other via passwordless SSH.
```
We should also add a security warning here that this passwordless SSH configuration may not be ideal for your environment. Please choose one that satisfies your organization's/client's security requirements.

nastena1606 replied:
Added

hqakhtar commented on docs/solutions/dr-pgbackrest-setup.md:
````diff
@@ -0,0 +1,481 @@
+   sudo yum install percona-pgbackrest
+   ```
+
+### Create the `pgBackRest` configuration file
````
This section is asking the user to perform far too many operations that should really be handled by our packages.

hqakhtar reviewed (commented):
Other than the inline feedback, here is some additional feedback:

Files to be removed/fixed:
- ha-setup-yum.md file shouldn't be included in this PR
- mkdocs-base.yml has mentions of 11.14. These should be included in a separate PR.

Overall, this seems like a fairly basic tutorial on pgbackrest. It doesn't provide any insights into any pitfalls, configuration parameters, or best practices even.

nastena1606 committed:
DISTPG-364 Disaster recovery docs

new file:   docs/_images/diagrams/DR-architecture.png
	new file:   docs/solutions/backup-recovery.md
	new file:   docs/solutions/dr-pgbackrest-setup.md
	modified:   mkdocs-base.yml

nastena1606 commented:
@EngineeredVirus, thank you for the feedback. The source for this document is the article provided by outsource authors and confirmed internally before it was to be added to the docs.
I've amended the docs to address your comments, please review.

hqakhtar reviewed (approved):
LGTM
